### ISSUE TITLE
Guava compile dependencies on runtime classpath

### ISSUE THREAD

**Issue by robobario** [linkedin/cruise-control#1934]
Hi, we build/run cruise-control and noticed that some guava compile dependencies are on the runtime path. As a user we'd like the minimum set of dependencies on the classpath. We currently exclude them in our build and wondered if it was worth doing the same upstream.

They are:
```
com.google.code.findbugs:jsr305
com.google.errorprone:error_prone_annotations
com.google.j2objc:j2objc-annotations
org.checkerframework:checker-qual
```

https://github.com/google/guava/wiki/UseGuavaInYourBuild#what-about-guavas-own-dependencies

Thank you

**Comment by CCisGG** [comment]
Hi @robobario , I think the intent is reasonable at least for your use cases. I'm concerned about what if in the future people make some changes that needs those dependencies?

**Comment by CCisGG** [comment]
I feel it's still safe to leave it there, unless it would affect performance or will add notable complexity during building or runtime.

**Comment by robobario** [comment]
True, it's safest to leave it since there's caveats around reflective access, maybe it'd break something in future. Thanks
